src/api/java/com/minecolonies/api/util/InventoryUtils.java

```diff
@@ -179,6 +180,25 @@ public static int findFirstSlotInItemHandlerWith(@NotNull final IItemHandler ite
         //throw new IllegalStateException("Item "+targetItem.getTranslationKey() + " not found in ItemHandler!");
     }
 
+    /**
+     * Shrinks a specific stack in an item handler.
+     * @param itemHandler the handler..
+     * @param itemStackSelectionPredicate the predicate..
+     * @return true if successful.
+     */
+    public static boolean shrinkItemCountInItemHandler(final IItemHandler itemHandler, @NotNull final Predicate<ItemStack> itemStackSelectionPredicate)
+    {
+        for (int slot = 0; slot < itemHandler.getSlots(); slot++)
+        {
+            if (itemStackSelectionPredicate.test(itemHandler.getStackInSlot(slot)))
```

Comment:
Insert an none empty check here. It makes no sense to call the predicate with an Empty stack when we are shrinking.